### PR TITLE
Dereference timers in node.js so that the process may exit when finished

### PR DESCRIPTION
**Pull request by danthegoodman**
Using the package in a node script is preventing the process from exiting because the reload interval timer has not been marked with `unref`. This PR resolves that issue.

For testing, I removed the `--exit` flag that had been previously been forcing mocha to exit immediately even if timers or servers were still open. This also required all servers and timers to in tests to be unrefed as well.

#### Checklist

- [x] only relevant code is changed (make a diff before you submit the PR)
- [x] run tests `npm run test`
- [x] tests are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/i18next/.github/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/http.spec.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ describe(`http backend using ${hasXMLHttpRequest() ? 'XMLHttpRequest' : 'fetch'}
             errO = err
             dataO = data
             resolve(true)
-            setTimeout(() => reject(new Error('timeout')), 1500)
           })
+          setTimeout(() => reject(new Error('timeout')), 1500).unref()
```
Comment (danthegoodman) [inline test/http.spec.js:45]: I moved this line so that it behaves as a proper timeout check against the backend.read call.

**Comment by danthegoodman** [comment]
I didn't update the `i18nextHttpBackend.js` and `….min.js` files because there were more changes in the files than just my targeted changes. I believe this is because the package-lock.json isn't in the repo, which means the babel/browserify packages have probably drifted slightly from the last run.

**Comment by adrai** [comment]
thank you, it's included in v2.5.2

**Comment by danthegoodman** [comment]
wow! So quick! Thank you 🎉

**Pull request merged**